### PR TITLE
Aidan sullivan issue13

### PR DESCRIPTION
**Pull request by codemyce**
# Summary

Simple route added to implement issue #13 . Basically collects the notes field a the provided client. Client is provided either by their email or their id.

Notes: Modified routes for /clients/<id> and /clients
Someone previously merged into master a commit that modified the Client model and broke these features. Accidentally included my fix in this commit.

## Test Plan

Testing the route host/notes/<id/email>
Tested them by having get requests for an email in the airtable, with an email not in the airtable, with an id in the airtable, an invalid id, and random junk.

## Related Issues

Which issues does this PR resolve/work on?
Implements issue #13

### CODE REVIEW

File: backend/app/main/views.py [hunk 1]
```diff
@@ -68,5 +69,40 @@ def get_a_client(id):
...
+    email_input = re.search(regex, id)
+
+    if (email_input):
+        # id is an email. Collect client notes by email
+        response = requests.get(
+            "https://api.airtable.com/v0/appw4RRMDig1g2PFI/Clients?filterByFormula=SEARCH('{}'".format(id) + ", {Client Email})",
+            headers={"Authorization": str(os.environ.get("API_KEY"))},
+        )
+    else:
+        # Not an email, assume it's an id
```
Comment (tedspare) [inline backend/app/main/views.py]: Could we use regex to check if this is a valid id?

File: backend/app/main/views.py [hunk 2]
```diff
@@ -68,5 +69,40 @@ def get_a_client(id):
...
-    else: 
-        return "This client does not exist in the database."
\ No newline at end of file
+    else:
+        return "This client does not exist in the database."
+
+# Gets list of client notes based on clientid or email
+@main.route("/notes/<id>", methods=["GET"])
+def get_client_notes(id):
+    # Regex used to check is string input is an email address
```
Comment (tedspare) [inline backend/app/main/views.py]: Minor typo, is -> if

File: backend/app/main/views.py (hunk 1 above)
Comment (codemyce) [reply backend/app/main/views.py]: Will do

File: backend/app/main/views.py (hunk 2 above)
Comment (codemyce) [reply backend/app/main/views.py]: Will correct

File: backend/app/main/views.py [hunk 3]
```diff
@@ -98,10 +100,52 @@ def get_a_client(id):
...
+    regex = '^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$'
+    email_input = re.search(regex, id)
+
+    regex = '^[A-Za-z0-9_()]{17}$'
+    id_input = re.search(regex, id)
+
+    if (email_input):
+        # id is an email. Collect client notes by email
+        response = requests.get(
+            "https://api.airtable.com/v0/appw4RRMDig1g2PFI/Clients?filterByFormula=SEARCH('{}'".format(id) + ", {Client Email})",
```
Comment (tedspare) [inline backend/app/main/views.py]: Should this be `.format(id)` or `.format(email_input)`?

**Pull request merged**